### PR TITLE
New Ctor for And and Or classes and increase test coverage

### LINKED ISSUE #13: And: ctor for passing an object and apply a list of functions to it.
We need a constructor on And.

I want to do the following:
```csharp
string s = "AB"
new And(
s,
str => str.Contains("A"),
str => str.Contains("B"),
str => str.Contains("C")).Value() //should be false
```

Or a solution with a different object

### LINKED ISSUE #161: Increase test coverage of OrTests
### Bug Report or Feature Request (mark with an x)
- [ ] bug report -> please search issues before submitting
- [ ] feature request,
- [x] Improvement

### Expected Behavior
All ctors are tested

### Actual Behavior
not all ctors are tested

### PR DESCRIPTION
**Pull request by ICACLE**
### Please check if the PR fulfills these requirements
- [x] I made sure that my code builds
- [x] I merged the master into this branch before pushing
- [x] The commit message follows our guidelines
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

### What kind of change does this PR introduce?** (check one with "x")
- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Other... Please describe:

### What is the current behavior? (You can also link to an open issue here)
closes #13
closes #161

### What is the new behavior?
Issues solved.

### Does this PR introduce a breaking change? (check one with "x")
- [ ] Yes
- [x] No

###  Other information

### CODE REVIEW

File: src/Yaapii.Atoms/Scalar/Or.cs [hunk 1]
```diff
@@ -65,6 +65,13 @@ public Or(IFunc<In, Boolean> func, params In[] src) : this(func, new EnumerableO
             )
         { }
 
+        /// <summary> True if any functions return true with given input value </summary>
+        /// <param name="value"> Input value wich will executed by all given functions </param>
+        /// <param name="functions"> Functions wich will executed with given input value </param>
+        public Or(In value, params Func<In, bool>[] functions)
+            : this(tValue => new Or(new Mapped<Func<In, bool>, bool>(tFunc => tFunc.Invoke(tValue), functions)).Value(), value)
```
Comment (Meerownymous) [inline src/Yaapii.Atoms/Scalar/Or.cs]: please remove the t prefix and fix indentation (this has to be on the right side)

File: tests/Yaapii.Atoms.Tests/Scalar/OrTests.cs [hunk 2]
```diff
@@ -44,6 +43,69 @@ public void TrueOrGenScalar()
                     ).Value() == true
             );
         }
-    }
-}
 
+        [Fact]
+        public void TestFunc()
```
Comment (Meerownymous) [inline tests/Yaapii.Atoms.Tests/Scalar/OrTests.cs]: Please make a name for the test which tells us what it tests

File: tests/Yaapii.Atoms.Tests/Scalar/OrTests.cs [hunk 3]
```diff
@@ -44,6 +43,69 @@ public void TrueOrGenScalar()
...
+        {
+            Assert.True(
+                    new Or<int>(
+                        input => input > 0,
+                        1, -1, 0
+                    ).Value());
+        }
+
+        [Fact]
+        public void TestFunc2()
```
Comment (Meerownymous) [inline tests/Yaapii.Atoms.Tests/Scalar/OrTests.cs]: Please make a name for the test which tells us what it tests

File: tests/Yaapii.Atoms.Tests/Scalar/OrTests.cs [hunk 4]
```diff
@@ -44,6 +43,69 @@ public void TrueOrGenScalar()
...
+        {
+            Assert.True(
+                    new Or<int>(
+                        input => input > 0,
+                        new List<int>() { 1, -1, 0 }
+                    ).Value());
+        }
+
+        [Fact]
+        public void TestIFunc()
```
Comment (Meerownymous) [inline tests/Yaapii.Atoms.Tests/Scalar/OrTests.cs]: Please make a name for the test which tells us what it tests

File: tests/Yaapii.Atoms.Tests/Scalar/OrTests.cs [hunk 5]
```diff
@@ -44,6 +43,69 @@ public void TrueOrGenScalar()
...
+                        new FuncOf<int, bool>(input => input > 0),
+                        -1, -2, -3
+                    ).Value());
+        }
+
+        [Theory]
+        [InlineData("DB", true)]
+        [InlineData("ABC", true)]
+        [InlineData("DEF", false)]
+        public void TestValueAndFunctionList(string value, bool expected)
```
Comment (Meerownymous) [inline tests/Yaapii.Atoms.Tests/Scalar/OrTests.cs]: Please choose a Fact about the class for a name

File: src/Yaapii.Atoms/Scalar/Or.cs (hunk 1 above)
Comment (ICACLE) [reply src/Yaapii.Atoms/Scalar/Or.cs]: Done

File: tests/Yaapii.Atoms.Tests/Scalar/OrTests.cs (hunk 5 above)
Comment (ICACLE) [reply tests/Yaapii.Atoms.Tests/Scalar/OrTests.cs]: Done

File: tests/Yaapii.Atoms.Tests/Scalar/OrTests.cs (hunk 3 above)
Comment (ICACLE) [reply tests/Yaapii.Atoms.Tests/Scalar/OrTests.cs]: Done

File: tests/Yaapii.Atoms.Tests/Scalar/OrTests.cs (hunk 2 above)
Comment (ICACLE) [reply tests/Yaapii.Atoms.Tests/Scalar/OrTests.cs]: Done

**Comment by ICACLE** [comment]
@Meerownymous Ready

**Comment by ICACLE** [comment]
@Meerownymous This PR seams to be forgotten...

**Comment by ICACLE** [comment]
@Meerownymous Ring!
@koeeenig Ring!

**Pull request merged**